### PR TITLE
Auto-enable async w/RestEasy Reactive, generate async void return types

### LINKED ISSUE #149: Methods return `void` instead of `CompletionStage<Void>` when reactive is enabled and OpenAPI operation has `204` return code
While using the apicurio-codegen with the `reactive` flag enabled, I have noticed that if an OpenAPI operation returns a `204 No Content` as HTTP response code, the generator will generate a resource method returning just `void`.
Let's take the following operation as an example ...

https://github.com/strimzi/strimzi-kafka-bridge/blob/main/src/main/resources/openapi.json#L287

apicurio-codegen will generate a method like this ...

```java
@Path("/{groupid}/instances/{name}/subscription")
@POST
@Consumes("application/vnd.kafka.v2+json")
void subscribe(@PathParam("groupid") String groupid, @PathParam("name") String name, Topics data);
```

Thinking about using this together with Quarkus RESTeasy Reactive, it sounds to me that if the method starts an async operation, the Quarkus framework responds to the HTTP client immediately with a `204` even if the async operation is not finished yet. At least this is my understanding.
Isn't it better to have apicurio-codegen generating methods returning `CompletionStage<Void>` in such a case?
This way, if an async operation starts, it has to complete the future before Quarkus can return the response to the HTTP client.

In our specific case, for example, the subscribe operation could take some time because it involves a Kafka consumer client to subscribe to topics in Kafka and it could even return an error.

Wdyt?

### PR DESCRIPTION
**Pull request by MikeEdgar**
Fixes #149

### CODE REVIEW

File: quarkus-extension/deployment/src/main/java/io/apicurio/openapi/server/generator/deployment/codegen/ApicurioOpenApiServerCodegen.java [hunk 1]
```diff
@@ -39,29 +44,53 @@ public boolean shouldRun(Path sourceDir, Config config) {
...
+                .getOptionalValue(getBasePackagePropertyName(), String.class)
+                .orElse(DEFAULT_PACKAGE));
+
+        projectSettings.setReactive(config
+                .getOptionalValue(CodegenConfig.getReactivePropertyName(), Boolean.class)
+                .orElseGet(() -> context.applicationModel()
+                            .getDependencies()
+                            .stream()
+                            .filter(dependency -> "quarkus-resteasy-reactive".equals(dependency.getArtifactId()))
+                            .anyMatch(dependency -> "io.quarkus".equals(dependency.getGroupId()))));
```
Comment (MikeEdgar) [inline quarkus-extension/deployment/src/main/java/io/apicurio/openapi/server/generator/deployment/codegen/ApicurioOpenApiServerCodegen.java:87]: This feels a bit brittle. I'm not finding anything better available to `CodeGenProvider` implementations. If others agree, I'll keep digging to see if there is a better approach.

Comment (EricWittmann) [reply quarkus-extension/deployment/src/main/java/io/apicurio/openapi/server/generator/deployment/codegen/ApicurioOpenApiServerCodegen.java:87]: shrug emoji  :)

Comment (MikeEdgar) [reply quarkus-extension/deployment/src/main/java/io/apicurio/openapi/server/generator/deployment/codegen/ApicurioOpenApiServerCodegen.java]: Ok - it turns out there is a collection of capabilities in the application model that is better than checking Maven coordinates. I've updated the PR to use that instead.

**Comment by MikeEdgar** [comment]
Third commit sets a fixed Quarkus version for the generated templates. Without it, the Quarkus Maven plugin seems to generate the project using the latest Quarkus version. The commit also uses a common variable for the CLI Quarkus version, bumping it to 3.0.4.Final (superseding #159)

**Pull request merged**